lib/run.bash

```diff
@@ -40,3 +40,12 @@ check_linked_containers_and_save_logs() {
     fi
   done
 }
+
+# docker-compose's -v arguments don't do local path expansion like the .yml
+# versions do. So we add very simply support, for the common and basic case.
+# 
+# "./foo:/foo" => "/buildkite/builds/.../foo:/foo"
+expand_relative_volume_path() {
+  local path="$1"
+  echo "${path/.\//$PWD/}"
```

Comment:
🤔How about something like:

```bash
if [[ -f $1 ]] ; then
  (cd $(dirname $1); pwd)
elsif [[ -d $1 ]] ; then
  (cd $1; pwd)
fi
```

Reply:
Oh, I just noticed that you were replacing it with `$PWD`. Why not just resolve the path?

Reply:
It just seemed to be the simplest thing that works? Happy to full resolve it using the trick you did though! I'll change it now.

Reply:
Yeah, fair enough. I reckon the fact that environment variables get expanded by the run eval is probably a bug. It makes me uncomfortable.

Reply:
> Whereas with -v I think it actually creates the path on the host filesystem if it doesn't exist yet?

I can confirm this is definitely true. If the host's path does not exist before the call to `-v`, then docker-compose will create it.

I think this looks fine as-is.